tensorflow_addons/losses/tests/kappa_loss_test.py

```diff
@@ -90,3 +90,22 @@ def test_config():
 def test_serialization():
     loss = WeightedKappaLoss(num_classes=3)
     tf.keras.losses.deserialize(tf.keras.losses.serialize(loss))
+
+
+def test_save_model(tmpdir):
+    model = tf.keras.models.Sequential(
+        [
+            tf.keras.layers.Input((256, 256, 3)),
+            tf.keras.layers.Dense(6, activation="softmax"),
+        ]
+    )
+    model.compile(optimizer="adam", loss=WeightedKappaLoss(num_classes=6))
+    model.save(str(tmpdir / "test.h5"))
+
+
+def test_dtype_config():
+    wkl = WeightedKappaLoss(num_classes=4, dtype=tf.float32)
```

Comment:
Is this the only loss in the repo where we handle `dtype` in the loss class?

Reply:
yes it is :-)

Reply:
is there a particular reason?

Reply:
Because it is very weird to pass `dtype` into losses. In core TF, the dtype of losses is determined either from inputs or from `backend.floatx()`. Consider the loss is going to pass into optimizers, which (mostly) creates the slots with the same dtype of variables, I think loss will be better to have either dtype `floatx()` or the one same with `y_pred/y_true`, so that the any computation in backward pass will not throw some errors like incompatible dtype.

Reply:
I meant Is there any particular reason why we are using it in this case?

Reply:
ohoh, sorry for misunderstanding :-(

Actually, I think no reason we do that. I would say `backend.floatx()` is the better solution. What do you think?

Reply:
If we don't break serialized users models I think It Is better

Reply:
Similar thoughts. That argument doesn't make sense as this should be inferred from the input dtype

Reply:
Ok, I revert the commit and use `tf.keras.backend.floatx()` instead.